### PR TITLE
Create new password feature done

### PR DESCRIPTION
**Pull request by lee-RoyGaseller**
If a user forgets their password they can now create a new one

### CODE REVIEW

File: src/main/java/codeu/model/data/User.java [hunk 1]
```diff
@@ -21,7 +21,7 @@
 public class User {
   private final UUID id;
   private final String name;
-  private final String passwordHash;
```
Comment (AaronGarcia97) [inline src/main/java/codeu/model/data/User.java]: This was in order to be able to modify the attributeright?

File: src/main/java/codeu/controller/NewPasswordServlet.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException, ServletException {
+        request.getRequestDispatcher("/WEB-INF/view/newPassword.jsp").forward(request, response);
+    }
+
+    @Override
+    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException, ServletException {
+        String newPassword = request.getParameter("newPassword");
+        String reTypedNewPassword = request.getParameter("reTypedNewPassword");
+        String username =request.getParameter("usernameVerification");
+
```
Comment (AaronGarcia97) [inline src/main/java/codeu/controller/NewPasswordServlet.java]: Remember to try and clean the user inputs from XSS and related attacks! Check the Jsoup library and methods we used at `ProfileServlet.java` and `AdminServlet.java` (doPost section). What it does is takes out html tags, etc!

File: src/main/java/codeu/model/data/User.java (hunk 1 above)
Comment (lee-RoyGaseller) [reply src/main/java/codeu/model/data/User.java]: yeah it was, sorry for the late response

File: src/main/java/codeu/controller/NewPasswordServlet.java (hunk 2 above)
Comment (lee-RoyGaseller) [reply src/main/java/codeu/controller/NewPasswordServlet.java]: okay will do, thanks!

Comment (lee-RoyGaseller) [reply src/main/java/codeu/controller/NewPasswordServlet.java]: okay I checked them out, what does clean the user inputs mean and what do the Jsoup methods actually do?

Comment (AaronGarcia97) [reply src/main/java/codeu/controller/NewPasswordServlet.java]: They take out html tags, if we don't do this a malicious user could input <script> doSomethingMalicious <script> and inject code into our server!

Comment (lee-RoyGaseller) [reply src/main/java/codeu/controller/NewPasswordServlet.java]: Okay I added the Jsoup method, is this correct? also shouldn't that be added to RegisterServlet.Java too?

Comment (AaronGarcia97) [reply src/main/java/codeu/controller/NewPasswordServlet.java]: Yeah, if I'm not mistaken the first if filters those tags and special characters using regex (short for regular expressions). :)

**Pull request merged**